### ISSUE TITLE
how pysam.PileUpColumn.nsegments is calculated from SAM or BAM file

### ISSUE THREAD

**Issue by pluralise** [pysam-developers/pysam#923]
Hi
Cross-posting this alongside my post in the pysam google user forum 
I am analysing some transcription sequencing data & the depth profile is important particularly at the start/end of transcripts - I want to be sure of how nsegments is calculated.
pysam.PileUpColumn.nsegments is useful for generating the depth across a region however I am unable to find out how the numbers for depth are calculated. 
Is anyone able to direct me to the code used in this calculation? Or to walk me through how the calculation is done?

I presume nsegments is generated from looking at all the CIGAR strings in the alignment as I can't see any more information in the output file from minimap2 that could be used. 
If this is the case, how are CIGAR strings that start or end in non-matches treated when calculating depth via nsegments? Are they used to calculate nsegments or ignored?

TL;DR 
pysam is great and could you advise me how it's working behind the scenes to generate PileUpColumn.nsegments from a SAM or BAM file.

Pysam User group ›
--
 
how pysam.PileUpColumn.nsegments is calculated from SAM or BAM file

Thank you,
Matt
